### PR TITLE
Cookbook artifacts match server behavior

### PR DESCRIPTION
**Pull request by danielsdeleo**
Cookbook Artifacts tests were not previously updated to test the actual behavior of cookbook artifacts that we implemented. In this PR I reverted some up factoring that shared test code between the two endpoints, since the behavior of the endpoints is different enough that code sharing between the tests would be a hassle. I then individually updated each file for the expected behavior. Note that much of the validation that the cookbooks endpoint does is lacking in cookbook artifacts, these tests have been set to 'skip' with a message explaining why.

@chef/lob

### CODE REVIEW

**Comment by tylercloke** [comment]
Generally agree with the PR description of the changes but didn't read all 5122 lines of the diff. Do you think we need to deep dive into all the changes here? Are there specific sections you'd like deeper review on?

**Comment by danielsdeleo** [comment]
The diff is probably more manageable if you skip the first 4 commits, which basically are a copy-past of the previous shared cookbook code into the cookbooks/ and cookbook_artifacts parts of the tree. Then the next commits update each endpoint for the actual behavior of the cookbook_artifacts endpoint one-by-one. I don't think a deep review is really needed, the tests are what they are.

**Comment by marcparadise** [comment]
I haven't deep-dived into all changes, but the shape of it looks good.   In the spirit of the tests being the tests and they'd fail if they were wrong, I'm okay with merging.

File: spec/api/cookbooks/named_filters_spec.rb [hunk 1]
```diff
@@ -13,12 +13,192 @@
...
+  context "with no cookbooks" do
+    let(:cookbooks) { {} }
+
+    should_respond_with_latest_cookbooks
+    should_respond_with_recipes_from_latest_cookbooks
+    should_respond_with_single_cookbook_not_found
+  end
+
+  # TODO: This would be a good candidate for a smoke test
+  # if there were a way to turn off testing against exact body responses.
```
Comment (marcparadise) [inline spec/api/cookbooks/named_filters_spec.rb]: Using the utils stuff it's really hard to customize at that level. I'm running into the same thing with customize search tests for API v1.

Comment (danielsdeleo) [inline spec/api/cookbooks/named_filters_spec.rb]: Actually, I think this comment is only relevant to OSS Chef Server 11. With CS 12, we can always test in an org so we could have smoke tests that write data if we want.

**Pull request merged**